frontend/src/routes/Search/Details/KubevirtPluginWrapper.tsx

```diff
@@ -2,18 +2,38 @@
 import React, { useContext, createContext } from 'react'
 import { PluginContext } from '../../../lib/PluginContext'
 import { KubevirtPluginData } from '../../../plugin-extensions/extensions/KubevirtContext'
+import { useAllClusters } from '../../Infrastructure/Clusters/ManagedClusters/components/useAllClusters'
+import { ResourceSearchLink } from './DetailsOverviewPage'
+import { K8sGroupVersionKind } from '@openshift-console/dynamic-plugin-sdk'
 
 // Define a default context
 const DefaultKubevirtPluginContext = createContext<KubevirtPluginData>({} as KubevirtPluginData)
 
-const ResourceLink = () => <div>Hello!!</div>
+const getResourceLinkOverride = (clusterName: string) => {
+  const ResourceLink = (props: { name?: string; groupVersionKind?: K8sGroupVersionKind }) => {
+    const { name = '', groupVersionKind } = props
+    if (!groupVersionKind) {
+      return null
+    }
+    const { version, kind } = groupVersionKind
+    return <ResourceSearchLink cluster={clusterName} apiversion={version} kind={kind} name={name} />
+  }
+
+  return ResourceLink
+}
+
+const KubevirtPluginWrapper = (props: { children: React.ReactNode; clusterName: string }) => {
+  const { children, clusterName } = props
+  const managedClusters = useAllClusters(true)
+  const isLocalCluster = managedClusters.find((cls) => cls.name === clusterName)?.labels?.['local-cluster'] === 'true'
+  if (isLocalCluster) {
+    return children
```

Comment:
You mentioned that we should not wrap the function if we are using a VM on the local cluster...are we ok with some deviation in flow with respect to the ResourceLink and other overrides from the SDK that will be tied in with ACM Search once overridden?
I'm realizing now that search and the OCP resource menus are visibly so different that it is a bit distracting to see.

Reply:
Good point! This is something we could ask UXD about if/when it is determined that we are going to actually implement some of this stuff.

Whether we go to the search details or the OpenShift resource details page, both of these are within-hub-console navigations, so in either case we are achieving the goal of not switching consoles.

If we want the user to be brought to the search details page even for the hub, we could still override `ResourceLink` in this case. But we can't override data functions that depend on the cluster proxy, because it cannot proxy to the hub itself.